### ISSUE TITLE
Clean up documentation output

### ISSUE THREAD

**Issue by adam-fowler** [swift-aws/aws-sdk-swift#322]
Currently it just strips out all the html tags and outputs that. Which means for some API documentation it just creates super long lines of text, which Xcode turns into a wall of text which is unreadable.

**Comment by RobbeRamon** [comment]
Hi Adam, can I provide you some help with this?

**Comment by adam-fowler** [comment]
@RobbeRamon yes that'd be great if you could look into this. Do you need some guidance on where to start?

**Comment by RobbeRamon** [comment]
I definitely could use some guidance given that this will be my first deep dive into this codebase

**Comment by adam-fowler** [comment]
I'll write something up tomorrow

**Comment by adam-fowler** [comment]
@RobbeRamon 
Here we go. Below is basic description of how the CodeGenerator works.
The CodeGenerator code can be found here https://github.com/soto-project/soto/tree/main/CodeGenerator. It uses a series of Mustache [templates](https://github.com/soto-project/soto/tree/main/CodeGenerator/mustache) to render each file. For each service file (api/shapes/errors/paginators) the CodeGenerator creates a context struct which is rendered with the relevant mustache template. Code for this can be found [here](https://github.com/soto-project/soto/blob/bc693b36e046d9f65e5a5cefe3036a37ebbbd19d/CodeGenerator/Sources/CodeGenerator/CodeGenerator.swift#L90). 

The contexts are created in [AWSService.swift](https://github.com/soto-project/soto/blob/main/CodeGenerator/Sources/CodeGenerator/AWSService.swift). There are a number of places where comments are generated inside the contexts. They all call `stripHTMLTags`. In general they are written as follows 
```swift
self.stripHTMLTags(docs)?.split(separator: "\n") ?? []
```
which strips all the HTML tags from the docs and splits the text on all newlines, giving you an array of `Substrings`

First thing I would do is break this off into a separate function and use this function throughout, not sure why I haven't done this already. You can then use this function as the base for all your work.

While developing you probably don't want to be re-creating all the services on every run it is a very expensive operation that creates 900,000 lines of code. If you are running of a laptop it will kill your battery. You can limit to just outputting one service by using the command line argument `--module` eg
```
soto-codegenerator --module s3 --format
```
I include the `--format` argument because the final generated code is generated with this argument (it runs swiftformat on the generated code). If you want to do comparisons with what is already there this'll make it easier.

I've added some thoughts on what needs to be done to clean up code in the main comment of this issue.

Given code generation is already an expensive operation trying to keep your code as optimal as you can. I don't really want to see code generation time double, but I also understand this work will have impact on code generation speed.
